app/controllers/uploads_controller.rb

```diff
@@ -0,0 +1,141 @@
+class UploadsController < ApplicationController
+  before_action :authenticate_user, except: [ :show, :download ]
+
+  def new
+    @upload = Upload.new
+  end
+
+  def create
+    begin
+      uploaded_file = file_params[:file]
+      (raise ActiveRecord::ParameterMissing, "Upload file") if uploaded_file.blank?
+      file_path, file_name, file_size, content_type = Upload.save_locally(uploaded_file, logged_in_user.id.to_s)
+      @upload = logged_in_user.uploads.new({
+        name: file_name,
+        file_path: file_path,
+        size: file_size,
+        is_public: file_params[:is_public].nil? ? false : file_params[:is_public],
+        content_type: content_type
+      })
+      respond_to do |format|
+        if @upload.save
+          format.html { redirect_to "/", notice: "File uploaded successfully." }
+          format.json { render json: { message: "File uploaded successfully.", upload: @upload }, status: :created }
+        else raise StandardError, "Unable to upload file"         end
+      end
+    rescue ActionController::ParameterMissing => e
+      handle_error(e.message, :bad_request)
+    rescue ActiveRecord::RecordNotFound => e
+      handle_error(e.message, :bad_request)
+    rescue StandardError => e
+      handle_error(e.message, :unprocessable_entity)
+    end
+  end
+
+  def index
+    @files = logged_in_user.uploads.order(created_at: :desc)
+    respond_to do |format|
+      format.html
+      format.json { render json: { files: @files } }
+    end
+  end
+
+  def update
+    begin
+      (raise ActionController::ParameterMissing, "fileId and is_public_updated are required.") if file_params[:is_public_updated].blank? || params[:file_id].blank?
+      file = logged_in_user.uploads.find_by(id: params[:file_id])
+      if file
+        is_public = file_params[:is_public_updated].to_i == 1 ? true : false
+        file.update(is_public: is_public)
+        respond_to do |format|
+          format.html { redirect_to "/", notice: "File visibility updated successfully." }
+          format.json { render json: { status: "Success" } }
+        end
+      else raise ActiveRecord::RecordNotFound, "File not found"       end
+    end
+  rescue ActiveRecord::RecordNotFound => e
+    handle_error(e.message, :bad_request)
+  rescue ActionController::ParameterMissing => e
+    handle_error(e.message, :bad_request)
+  rescue StandardError => e
+    handle_error(e.message, :unauthorized)
+  end
+
+  def destroy
+    permitted_params = params.permit(:file_id)
+    (raise ActionController::ParameterMissing, "fileId is required.") if permitted_params[:file_id].blank?
+    file_id = permitted_params[:file_id]
+    file = logged_in_user.uploads.find_by!(id: file_id)
+    (raise ActiveRecord::RecordNotFound, "Unable to delete") unless file.delete_upload
+    respond_to do |format|
+      file.destroy
```

Comment:
what happens if file is deleted from db deletion fails

Reply:
adding begin block
If it fails we will go to the exception catch